vnext/ReactUWP/Views/ViewViewManager.cpp

```diff
@@ -134,20 +134,26 @@ class ViewShadowNode : public ShadowNodeBase
     static_cast<FrameworkElementViewManager*>(GetViewManager())->RefreshTransformMatrix(this);
   }
 
-  winrt::com_ptr<ViewPanel> GetViewPanel()
+  winrt::com_ptr<ViewPanel> GetViewPanel(bool removeAllChildren = false)
   {
```

Comment:
This feels like an odd way to induce removal of the children. Why this instead of just embedding this in removeAllChildren()? Just the logic for getting the container?

Reply:
I didn't want to duplicate the logic to check for the control and the border.
But now that you mention it, including a param removeAllChildren on GetViewPanel does seem strange. I'll take another crack at this.

Reply:
It also might work to include more helper functions so you don't duplicate too much logic.

---
In reply to: [296325388](https://github.com/microsoft/react-native-windows/pull/2658#discussion_r296325388) [](ancestors = 296325388)